cpp/examples/tpch/utils.hpp

```diff
@@ -0,0 +1,438 @@
+#include <cudf/binaryop.hpp>
+#include <cudf/sorting.hpp>
+#include <cudf/transform.hpp>
+#include <cudf/reduction.hpp>
+#include <cudf/unary.hpp>
+#include <cudf/stream_compaction.hpp>
+#include <cudf/reduction.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+
+#include <rmm/cuda_device.hpp>
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+#include <rmm/mr/device/pool_memory_resource.hpp>
+#include <rmm/mr/device/owning_wrapper.hpp>
+
+
+/**
+ * @brief Create memory resource for libcudf functions
+ *
+ * @param pool Whether to use a pool memory resource.
+ * @return Memory resource instance
+ */
+std::shared_ptr<rmm::mr::device_memory_resource> create_memory_resource(bool pool)
+{
+  auto cuda_mr = std::make_shared<rmm::mr::cuda_memory_resource>();
+  if (pool) {
+    return rmm::mr::make_owning_wrapper<rmm::mr::pool_memory_resource>(
+      cuda_mr, rmm::percent_of_free_device_memory(50));
+  }
+  return cuda_mr;
+}
+
+/**
+ * @brief A class to represent a table with column names attached
+ */
+class table_with_cols {
+    public:
+        table_with_cols(
+            std::unique_ptr<cudf::table> tbl, std::vector<std::string> col_names) 
+                : tbl(std::move(tbl)), col_names(col_names) {}
+        cudf::table_view table() {
+            return tbl->view();
+        }
+        /**
+         * @brief Return the column view for a given column name
+         *
+         * @param col_name The name of the column
+         */
+        cudf::column_view column(std::string col_name) {
+            return tbl->view().column(col_id(col_name));
+        }
+        /**
+         * @param Return the column names of the table
+         */
+        std::vector<std::string> columns() {
+            return col_names;
+        }
+        /**
+         * @brief Translate a column name to a column index
+         *
+         * @param col_name The name of the column
+         */
+        cudf::size_type col_id(std::string col_name) {
+            CUDF_FUNC_RANGE();
+            auto it = std::find(col_names.begin(), col_names.end(), col_name);
+            if (it == col_names.end()) {
+                throw std::runtime_error("Column not found");
+            }
+            return std::distance(col_names.begin(), it);
+        }
+        /**
+         * @brief Append a column to the table
+         *
+         * @param col The column to append
+         * @param col_name The name of the appended column
+         */
+        std::unique_ptr<table_with_cols> append(std::unique_ptr<cudf::column>& col, std::string col_name) {
+            CUDF_FUNC_RANGE();
+            std::vector<std::unique_ptr<cudf::column>> updated_cols;
+            std::vector<std::string> updated_col_names;
+            for (size_t i = 0; i < tbl->num_columns(); i++) {
+                updated_cols.push_back(std::make_unique<cudf::column>(tbl->get_column(i)));
```

Comment:
This copies the columns of `this` to produce thew new table. Which is slightly strange semantics for something called `append`. Is that really what you want?